src/highdicom/sr/sop.py

```diff
@@ -180,67 +180,168 @@ def __init__(
+
+        ref_items = []
+        for study_uid, series_items in study_collection.items():
+            study_item = Dataset()
+            study_item.StudyInstanceUID = study_uid
+            study_item.ReferencedSeriesSequence = series_items
+            ref_items.append(study_item)
+
+        return ref_items
+
+    def _collect_predecessors(
+        self,
+        previous_versions: List[Dataset]
+    ) -> List[Dataset]:
+        """Collect predecessors of the SR document.
+
+        Parameters
+        ----------
+        previous_versions: List[pydicom.dataset.Dataset]
+            Metadata of instances that represent previous versions of the
+            SR document content
+
+        Returns
+        -------
+        List[pydicom.dataset.Dataset]
+            Items of the Predecessor Documents Sequence
+
+        """
+        collection: Dict[str, List[Dataset]] = defaultdict(list)
+        for pre in previous_versions:
+            pre_instance_item = Dataset()
+            pre_instance_item.ReferencedSOPClassUID = pre.SOPClassUID
+            pre_instance_item.ReferencedSOPInstanceUID = pre.SOPInstanceUID
+            key = (pre.StudyInstanceUID, pre.SeriesInstanceUID)
+            collection[key].append(pre_instance_item)
+        return self._create_references(collection)
+
+    def _collect_evidence(
+        self,
+        evidence: List[Dataset],
+        content: Dataset
+    ) -> Tuple[List[Dataset], List[Dataset]]:
+        """Collect evidence for the SR document.
+
+        Any `evidence` that is referenced in `content` via IMAGE or
+        COMPOSITE content items will be grouped together for inclusion in the
+        Current Requested Procedure Evidence Sequence and all remaining
+        evidence will be grouped for potential inclusion in the Pertinent Other
+        Evidence Sequence.
+
+        Parameters
+        ----------
+        evidence: List[pydicom.dataset.Dataset]
+            Metadata of instances that serve as evidence for the SR document
+            content
+        content: pydicom.dataset.Dataset
+            SR document content
+
+        Returns
+        -------
+        Tuple[List[pydicom.dataset.Dataset], List[pydicom.dataset.Dataset]]
+            Items of the Current Requested Procedure Evidence Sequence and the
+            Pertinent Other Evidence Sequence
+
+        Raises
+        ------
+        ValueError
+            When a SOP instance is referenced in `content` but not provided as
+            `evidence`
+
+        """  # noqa
+        references = find_content_items(
+            content,
+            value_type=ValueTypeValues.IMAGE,
+            recursive=True
+        )
+        references += find_content_items(
+            content,
+            value_type=ValueTypeValues.COMPOSITE,
+            recursive=True
+        )
+        ref_uids = [
+            ref.ReferencedSOPSequence[0].ReferencedSOPInstanceUID
+            for ref in references
+        ]
```

Comment:
Minor preference probably - but I would tend to make this a set instead of a list since order is unimportant, there may be duplicates references that only need to be in this container once, and it will be used for lots of look-up operations. That would also allow you to use set operations between `ref_uids` and `evd_uids` later, rather than another loop.

Reply:
Addressed with 8e6ad6259f7553cbd708bbc392d2b2f41df540ae